loadtest/perftest.sh

```
@@ -104,6 +134,15 @@ else
   say "   (py-spy not installed — skipping CPU profile; see README.md)"
 fi
 
+# --- resource sampler (CPU + RAM for every service, continuously)
+# One-off readings of `docker stats` disagreed by 20x within a single run,
+# so this samples kernel counters for the whole window and the report shows
+# the distribution rather than a spot value.
+if [ -x "$HERE/resource_sampler.sh" ]; then
+    "$HERE/resource_sampler.sh" "$OUTDIR" "$(( SECS + 10 ))" 2 >/dev/null 2>&1 &
+    RES_PID=$!
```

Shellcheck (0.11.0)

[warning] RES_PID appears unused. Verify use (or export if used externally).

(SC2034)